src/controllers/updatePassword.controller.ts

```diff
@@ -0,0 +1,42 @@
+import { Request, Response } from "express";
+import bcrypt from "bcryptjs";
+import User from "../database/models/User.model";
+
+export const updatePassword = async (req: Request, res: Response) => {
+  try {
+    const { currentPassword, newPassword, newPasswordConfirmation } = req.body;
+    if (
+      currentPassword.length < 8 ||
+      newPassword.length < 8 ||
+      newPasswordConfirmation.length < 8
+    ) {
+      throw new Error("password is vey short, try more than 8 characters");
+    }
```

Comment:
There is already a regex in the user.middleware.ts which you can just place in the route handler. No need for this.

Reply:
thanks, that is awesome I've looked at it